openml/_api_calls.py

```diff
@@ -159,9 +231,12 @@ def _parse_server_exception(
     try:
         server_exception = xmltodict.parse(response.text)
     except Exception:
-        raise OpenMLServerError(
-            'Unexpected server error when calling {}. Please contact the developers!\n'
-            'Status code: {}\n{}'.format(url, response.status_code, response.text))
+        if response.status_code == 414:
```

Comment:
Does it not make sense to add this before the try/except block? I don't know the details, but sounds like you would always assume the parsing to fail if the response had code `414`.

Reply:
> Does it not make sense to add this before the try/except block?

Yes, it does.

> but sounds like you would always assume the parsing to fail if the response had code 414.

No, a lot of things can happen with the server output...